src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt

```diff
@@ -222,20 +222,18 @@ class ClientService constructor(peerPort: Int? = findFreePort()) : IClientServic
         fun dispatchMessage(clientMessage: ClientMessage, senderPeerAddress: PeerAddress) {
             val recipientSessionId = clientMessage.recipientSessionId!!
             val senderSessionId = clientMessage.senderSessionId!!
+
             emitterMap[recipientSessionId]?.let {
                 logger.info { "message accepted, found emitter for $recipientSessionId" }
                 it(ExternalClient(senderSessionId, senderPeerAddress, peer), clientMessage)
-            } ?: run {
-                logger.info { "message discarded (no registered emitter for session id $recipientSessionId" }
-            }
+            } ?: logger.info { "message discarded (no registered emitter for session id $recipientSessionId" }
         }
 
         /** Only dispatch a message if it's actually one of our own messages */
-        fun tryDispatchingMessage(messageDto: Any?, senderPeerAddress: PeerAddress): Any {
+        fun tryDispatchingMessage(messageDto: Any?, senderPeerAddress: PeerAddress) {
             logger.info { "got message $messageDto" }
-            return if (messageDto is ClientMessage) {
```

Comment:
With this change, I get a TomP2P error. Seems like we have to return something here, I'm not sure if it needs to be the message or just *something*.

The error I get is a NotSerializableException: kotlin.Unit. Seems like whatever we're returning here needs to be serializable. Why? I have no clue...

Also, it seems that in the line below, there's an implicit return (tryDispatchingMessage(...))

Reply:
I think the other changes make sense, just not removing this return